File: src/Helpers/AtFileHelpers.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

public class AtFileHelpers
{
    public static string ExpandAtFileValue(string atFileValue, INamedValues? values = null)
    {
        if (atFileValue.StartsWith("@") && FileHelpers.FileExists(atFileValue[1..]))
        {
            return FileHelpers.ReadAllText(atFileValue[1..]);
        }
        else if (atFileValue.StartsWith("@") && ConsoleHelpers.IsStandardInputReference(atFileValue[1..]))
        {
            return string.Join('\n', ConsoleHelpers.GetAllLinesFromStdin());
        }
        else if (atFileValue.StartsWith("@"))
        {
            string key = atFileValue[1..];

            // First check the provided values if any
            if (values != null && values.Contains(key))
            {
                string? result = values.Get(key);
                if (result != null)
                {
                    return result;
                }
            }

            // Then check ConfigStore
            var configValue = ConfigStore.Instance.GetFromAnyScope(key);
            if (!configValue.IsNotFoundNullOrEmpty())
            {
                return configValue.AsString() ?? string.Empty;
            }

            // Then check embedded files
            var embeddedFileExists = EmbeddedFileExists(key);
            if (embeddedFileExists)
            {
                return ReadEmbeddedStream(key)!;
            }
        }

        return atFileValue;
    }

    private static string? ReadEmbeddedStream(string key)
    {
        var value = EmbeddedFileHelpers.ReadEmbeddedStream(key);
        if (value != null) return value;

        value = EmbeddedFileHelpers.ReadEmbeddedStream($"prompts.{key}");
        if (value != null) return value;

        value = EmbeddedFileHelpers.ReadEmbeddedStream($"prompts.{key}.md");
        if (value != null) return value;

        return null;
    }

    private static bool EmbeddedFileExists(string key)
    {
        return EmbeddedFileHelpers.EmbeddedStreamExists(key)
            || EmbeddedFileHelpers.EmbeddedStreamExists($"prompts.{key}")
            || EmbeddedFileHelpers.EmbeddedStreamExists($"prompts.{key}.md");
    }
}

File: src/commonjs/Helpers/AtFileHelpers.ts
import { INamedValues } from './INamedValues';
import { FileHelpers } from './FileHelpers';
import { ConsoleHelpers } from './ConsoleHelpers';
import { ConfigStore } from '../Configuration/ConfigStore';
import { EmbeddedFileHelpers } from './EmbeddedFileHelpers';

export class AtFileHelpers {
    public static expandAtFileValue(atFileValue: string, values?: INamedValues): string {
        if (atFileValue.startsWith('@') && FileHelpers.fileExists(atFileValue.slice(1))) {
            return FileHelpers.readAllText(atFileValue.slice(1));
        } else if (atFileValue.startsWith('@') && ConsoleHelpers.isStandardInputReference(atFileValue.slice(1))) {
            return ConsoleHelpers.getAllLinesFromStdin().join('\n');
        } else if (atFileValue.startsWith('@')) {
            const key = atFileValue.slice(1);

            // First check the provided values if any
            if (values && values.Contains(key)) {
                const result = values.Get(key);
                if (result !== null) {
                    return result;
                }
            }

            // Then check ConfigStore
            const configValue = ConfigStore.instance.getFromAnyScope(key);
            if (!configValue.isNotFoundNullOrEmpty()) {
                return configValue.asString() ?? '';
            }

            // Then check embedded files
            const embeddedFileExists = AtFileHelpers.embeddedFileExists(key);
            if (embeddedFileExists) {
                return AtFileHelpers.readEmbeddedStream(key)!;
            }
        }

        return atFileValue;
    }

    private static readEmbeddedStream(key: string): string | null {
        let value = EmbeddedFileHelpers.readEmbeddedStream(key);
        if (value !== null) return value;

        value = EmbeddedFileHelpers.readEmbeddedStream(`prompts.${key}`);
        if (value !== null) return value;

        value = EmbeddedFileHelpers.readEmbeddedStream(`prompts.${key}.md`);
        if (value !== null) return value;

        return null;
    }

    private static embeddedFileExists(key: string): boolean {
        return EmbeddedFileHelpers.embeddedStreamExists(key)
            || EmbeddedFileHelpers.embeddedStreamExists(`prompts.${key}`)
            || EmbeddedFileHelpers.embeddedStreamExists(`prompts.${key}.md`);
    }
}